test/over_react/component/lazy_test.dart

```diff
@@ -0,0 +1,663 @@
+
+import '../component_declaration/builder_integration_tests/new_boilerplate/function_component_test.dart';
+import '../util/prop_conversion_test.dart';
+import '../util/ref_test_cases.dart';
+
+part 'lazy_test.over_react.g.dart';
+
+main() {
+  enableTestMode();
+
+  group('lazy', () {
+    // The fallback and the lazy component render the same thing...
+    uiPropsTest(
+      uiFunction<UiProps>(
+        (props) => (Suspense()..fallback = (Dom.div()..addTestId('testId3'))('id: ${props.id}'))(
+          lazy(
+              () async => uiFunction<UiProps>(
+                    (props2) => (Dom.div()..addTestId('testId3'))('id: ${props.id}'),
+                    UiFactoryConfig(),
+                  ),
+              UiFactoryConfig())()(),
+        ),
+        UiFactoryConfig(),
+      ),
+    );
+
+    group('does not throw an error when', () {
+      test('config is null, IF it only expects GenericUiProps', () {
+        testLazy() => lazy(
+            () async => uiFunction<UiProps>(
+                  (props) => (Dom.div()..addTestId('testId3'))('id: ${props.id}'),
+                  UiFactoryConfig(),
+                ),
+            null);
+        expect(testLazy, returnsNormally);
+        expect(testLazy(), isA<UiFactory<UiProps>>());
+      });
+    });
+
+    group('throws an error when', () {
+      test('props factory is not provided when using custom props class', () {
+        expect(
+            () => lazy(
+                  () async => uiFunction<TestProps>(
+                    (props) => Dom.div()(),
+                    UiFactoryConfig(displayName: 'Foo'),
+                  ),
+                  UiFactoryConfig(displayName: 'Foo'),
+                ),
+            throwsArgumentError);
+      });
+
+      test('config not the correct type', () {
+        expect(
+            () => lazy(
+                  () async => uiFunction<UiProps>(
+                    (props) => (Dom.div()..addTestId('testId3'))('id: ${props.id}'),
+                    UiFactoryConfig(),
+                  ),
+                  'foo',
+                ),
+            throwsArgumentError);
+      });
+    });
+
+    group('initializes the factory variable with a function', () {
+      test('that returns a new props class implementation instance', () {
+        final instance = lazy(() async => TestDart, _$TestDartConfig)();
+        expect(instance, isA<TestDartProps>());
+        expect(instance, isA<Map>());
+      });
+
+      test('that returns a new props class implementation instance backed by an existing map', () {
+        Map existingMap = {'stringProp': 'test'};
+        final factory = lazy(() async => TestDart, _$TestDartConfig);
+        final props = factory(existingMap);
+
+        expect(props.stringProp, equals('test'));
+
+        props.stringProp = 'modified';
+        expect(props.stringProp, equals('modified'));
+        expect(existingMap['stringProp'], equals('modified'));
+      });
+    });
+
```

Comment:
It looks like there isn't a basic test in here that the lazy component actually renders, though it is tested in the wrapper cases, and also in suspense_test.dart. 

Would you mind either adding a comment in here that mentions that, or adding a test here?

Reply:
Added one under the `does not throw an error when` group called `wrapped in suspense (renders correctly)` that checks that there were no errorboundary errors and that the component is displaying correctly. Its also kinda asserted in `renders a component from end to end, successfully reading props via typed getters` from the `uiPropsTests`